src/Umbraco.Web.UI.Client/src/common/directives/components/umbicon.directive.js

```diff
@@ -43,13 +34,19 @@ This format is only used in the iconpicker.html
                 svgString: "=?"
             },
 
-            link: function (scope) {
-                
+            link: function (scope, element) {
                 if (scope.svgString === undefined && scope.svgString !== null && scope.icon !== undefined && scope.icon !== null) {
-                    var icon = scope.icon.split(" ")[0]; // Ensure that only the first part of the icon is used as sometimes the color is added too, e.g. see umbeditorheader.directive scope.openIconPicker
+                    const observer = new IntersectionObserver(_lazyRequestIcon, {rootMargin: "100px"});
```

Comment:
What does this mean? I can not decipher it 😅

Reply:
No worries, the IntersectionObserver is a sweet native API supported by all modern browsers that will fire a callback when the observed DOM element comes into view. The `rootMargin: "100px"` essentially means that it will add an additional 100px around the element when it triggers the callback.

The `_lazyRequestIcon`  checks to see if the element is intersecting with the view according to the options, and if it sees that is it, it will fetch make the request, and then disconnect the observer as it is no longer needed.